torchvision/prototype/transforms/_geometry.py

```diff
@@ -3,12 +3,12 @@
 import torch
 from torch.nn.functional import interpolate
 from torchvision.prototype.datasets.utils import SampleQuery
-from torchvision.prototype.features import BoundingBox, Image, Label
+from torchvision.prototype.features import BoundingBox, Image, Label, KeyPoint
 from torchvision.prototype.transforms import Transform
 
 
 class HorizontalFlip(Transform):
-    NO_OP_FEATURE_TYPES = {Label}
+    NO_OP_FEATURE_TYPES = {Label, KeyPoint}
```

Comment:
Keypoint should be aware of how to flip itself.

This is our current implementation for COCO: https://github.com/pytorch/vision/blob/d5a22a852f178b181a2077e01a14eae9e25a8ff7/references/detection/transforms.py#L10-L17

basically, `left` becomes `right` and vice-versa, so this needs to be taken into account as well (apart from just the coordinate flip).

This hints to having an extra meta-data associated with keypoints that can be applicable in those use-cases.
This is what I came up with back in [`maskrcnn-benchmark`](https://github.com/facebookresearch/maskrcnn-benchmark/blob/57eec25b75144d9fb1a6857f32553e1574177daf/maskrcnn_benchmark/structures/keypoint.py#L97-L130), but which can probably be improved upon / generalized

Reply:
We are in the process of revamping the prototype transforms and thus I'm not going to add transforms for the `KeyPoint`'s in this PR. Still, you got a point about the descriptions.

To achieve these automatic conversion, we would need to store some kind of symmetry information about the individual keypoints. In the example of persons or faces we only have horizontal symmetry, but I don't think it is far fetched to also consider other symmetries such as vertical. Thus, I'm hesitant to add meta data to the feature that is only valid for a single transformation. 

Is the order of the keypoints important for further processing? If not, maybe we can take the easy way out and simply invalidate the descriptions in transformations like `horizontal_flip`?

Reply:
Yes, the order of the keypoints is super important. Indeed, we train models to predict left-right elements, and their order in the vector indicates if they are left-right.

If we don't carry this information properly, it will be impossible to apply those transformations to the object, or they will be wrong.

And I strongly believe we should take transformations into account when defining what should be in the Feature. This is one of the main reasons why we have Feature to begin with.

I believe we should add enough metadata that geometric transformations are well-defined. `image_size` is an easy case, and we do take this information into account as well in the `(x0, y0, x1, y1)` bounding-box world (we know that 3rd element is x1, and this changes how we perform flip).

Reply:
I've pushed a new commit that handles this.